src/Compilers/CSharp/Portable/Syntax/SyntaxFactory.cs

```diff
@@ -1693,9 +1693,19 @@ public static NameSyntax ParseName(string text, int offset = 0, bool consumeFull
         /// <summary>
         /// Parse a TypeNameSyntax node using the grammar rule for type names.
         /// </summary>
-        public static TypeSyntax ParseTypeName(string text, int offset = 0, bool consumeFullText = true)
+        // Backcompat overload, do not remove
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public static TypeSyntax ParseTypeName(string text, int offset, bool consumeFullText)
         {
-            using (var lexer = MakeLexer(text, offset))
+            return ParseTypeName(text, offset, options: null, consumeFullText);
+        }
+
+        /// <summary>
+        /// Parse a TypeNameSyntax node using the grammar rule for type names.
+        /// </summary>
+        public static TypeSyntax ParseTypeName(string text, int offset = 0, ParseOptions? options = null, bool consumeFullText = true)
```

Comment:
>ParseOptions [](start = 76, length = 12)

Should this be ```CSharpParseOptions``` as in ```ParseCompilationUnit```?

Reply:
We seem to be inconsistent as to whether we take a ParseOptions or a CSharpParseOptions, but given that this (and the rest where we take a `ParseOptions`) just casts immediately I'll make a `CSharpParseOptions`.

---
In reply to: [432679991](https://github.com/dotnet/roslyn/pull/44564#discussion_r432679991) [](ancestors = 432679991)